src/fiber/fiber.ml

```diff
@@ -393,8 +389,8 @@ module Ivar = struct
     | Full _ -> failwith "Fiber.Ivar.fill"
     | Empty q ->
       t.state <- Full x;
-      K.run_queue q x;
-      k ()
+      k ();
+      K.run_queue q x
```

Comment:
Now that I think about it, why don't we just get rid of `k` altogether and return `unit`:

```ocaml
val fill : 'a t -> 'a -> unit
```

This will make it work exactly, like async's ivar and I don't think we lose much. Code like this:

```ocaml
let*() = fill ivar a in
foo ()
```

can almost certainly be written as:
```ocaml
let* () = foo () in
fill ivar a;
```

wdyt?

Reply:
it's not the same. Even if the waiters are not awoken immediately, the ivar is still marked as filled immediately in the first case.

Reply:
alright, let's keep it this way.

Reply:
To make it work, we'd need `fill` to move the waiters to a central queue and wake them up in `Fiber.run`.

However, I'm hoping that we can one day revert to the old implementation where the execution context is passed functionally rather than being stored in a global variable. Currently, the only reason it has to be in a global variable is because of `Fiber.Var.get`. But once we have fiberized everything, maybe we won't need it anymore.